db/migrate/20160329184508_create_gem_download_count_index.rb

```diff
@@ -0,0 +1,5 @@
+class CreateGemDownloadCountIndex < ActiveRecord::Migration
+  def change
+    add_index :gem_downloads, [:version_id, :rubygem_id, :count]
```

Comment:
- Having `count` will avoid a data page hit. Which is good.
- But since `count` will be changing a lot, the index will be changing a lot. And potentially render the table statistics invalid. So the index would not be very effective, and the data page hit may actually be quicker than dealing with a stale index.
- The `count` column may slow down updates as well.
- `version_id` is almost always be `0` - which is not good at the front of an index. Putting values that are the most different at the front makes for better data distribution (You can challenge this. this may no longer be valid) And yes, you can only use indexes from left to right, so this rule of thumb is not always applicable.
- Making the index `where version_id = 0` would probably be your best bet. (not `schema.rb` friendly - in rails discussions) - but in terms of postgres - it makes a BIG difference. especially for a query such as this.

e.g.:
```sql
create index gem_downloads_rubygem_id on gem_downloads(rubygem_id) where version_id=0
```